### PR TITLE
Release 2023-01-11

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## Test Plan
https://docs.google.com/spreadsheets/d/1xagjgvjHX1vO-tXv9WW32iNbCRSOxMAZlif1wdhnY3E/edit#gid=333071376

## Social Release Jan 2023 - Integration Status
https://docs.google.com/spreadsheets/d/1qqseVUBMYp63xcadSXsTz_caRFQ4VOcBV1Evqa3EfKQ/edit#gid=802671791

## Test Link
https://play.decentraland.org/?renderer-branch=release/release-20230111&ENABLE_AVATAR_OUTLINER&ENABLE_NEW_FRIEND_REQUESTS&ENABLE_NEW_AVATAR_FLOW

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 1]
```diff
@@ -11,32 +13,55 @@ namespace DCL.Social.Passports
...
-            view.OnClickBuyNft += (wearableId) => OnClickBuyNft?.Invoke(wearableId);
+            view.OnClickBuyNft += (wearableId, wearableType) => OnClickBuyNft?.Invoke(wearableType is "name" or "parcel" or "estate" ? currentUserId : wearableId, wearableType);
+            view.OnClickCollectibles += () => OnClickCollectibles?.Invoke();
         }
 
         public void UpdateWithUserProfile(UserProfile userProfile) => UpdateWithUserProfileAsync(userProfile).Forget();
 
+        public void CloseAllNFTItemInfos() => view.CloseAllNFTItemInfos();
+
         private async UniTaskVoid UpdateWithUserProfileAsync(UserProfile userProfile)
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:62]: Add `CancellationToken`: it should be Canceled when `View` is closed.
In general you can't have asynchronous code without cancellation tokens

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 2]
```diff
@@ -11,32 +13,55 @@ namespace DCL.Social.Passports
...
+            this.namesService = namesService;
+            this.landsService = landsService;
+            this.userProfileBridge = userProfileBridge;
             this.dataStore = dataStore;
-            view.OnClickBuyNft += (wearableId) => OnClickBuyNft?.Invoke(wearableId);
+            view.OnClickBuyNft += (wearableId, wearableType) => OnClickBuyNft?.Invoke(wearableType is "name" or "parcel" or "estate" ? currentUserId : wearableId, wearableType);
+            view.OnClickCollectibles += () => OnClickCollectibles?.Invoke();
         }
 
         public void UpdateWithUserProfile(UserProfile userProfile) => UpdateWithUserProfileAsync(userProfile).Forget();
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:58]: No reason to launch a separate flow as you call this method from another async flow. Make it return `UniTask` and await from the caller accordingly

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 3]
```diff
@@ -45,9 +70,11 @@ private async UniTaskVoid UpdateWithUserProfileAsync(UserProfile userProfile)
             {
                 string filteredDescription = await FilterContentAsync(userProfile.description);
                 view.SetDescription(filteredDescription);
+                view.SetHasBlockedOwnUser(userProfile.IsBlocked(ownUserProfile.userId));
                 await LoadAndDisplayEquippedWearablesAsync(userProfile);
             }
         }
+
         private async UniTask LoadAndDisplayEquippedWearablesAsync(UserProfile userProfile)
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:78]: Add cancellation token

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 4]
```diff
@@ -45,9 +70,11 @@ private async UniTaskVoid UpdateWithUserProfileAsync(UserProfile userProfile)
...
         {
             CancellationToken ct = new CancellationToken();
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:80]: Nonsense

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 5]
```diff
@@ -59,6 +86,8 @@ private async UniTask LoadAndDisplayEquippedWearablesAsync(UserProfile userProfi
                     cachedAvatarEquippedWearables = new HashSet<string>(userProfile.avatar.wearables);
                     LoadAndShowOwnedWearables(userProfile);
                     LoadAndShowOwnedEmotes(userProfile);
+                    LoadAndShowOwnedNamesAsync(userProfile).Forget();
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:89]: Don't launch a separate flow, use `UniTask.WhenAll` instead.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 6]
```diff
@@ -59,6 +86,8 @@ private async UniTask LoadAndDisplayEquippedWearablesAsync(UserProfile userProfi
...
+                    LoadAndShowOwnedLandsAsync(userProfile).Forget();
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:90]: Don't launch a separate flow, use UniTask.WhenAll instead.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 7]
```diff
@@ -77,21 +107,57 @@ private void LoadAndShowOwnedWearables(UserProfile userProfile)
...
+            if (response.success)
+                landsResult = response.response.Lands.ToArray();
+            else
+                Debug.LogError("Error requesting lands lambdas!");
+
+            view.SetCollectibleLands(landsResult);
+            view.SetCollectibleLandsLoadingActive(false);
+        }
+
         private async UniTask<string> FilterContentAsync(string filterContent) =>
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:161]: Please change it to accept `CancellationToken`, also please `Filter` to accept it as well and handle properly: it should not take long

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 8]
```diff
@@ -77,21 +107,57 @@ private void LoadAndShowOwnedWearables(UserProfile userProfile)
...
                                  .Then(emotes =>
                                   {
                                       WearableItem[] emoteItems = emotes.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT).ToArray();
                                       view.SetCollectibleEmotes(emoteItems);
+                                      view.SetCollectibleEmotesLoadingActive(false);
                                   })
                                  .Catch(Debug.LogError);
         }
 
+        private async UniTask LoadAndShowOwnedNamesAsync(UserProfile userProfile)
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:128]: Add `CancellationToken`

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs [hunk 9]
```diff
@@ -77,21 +107,57 @@ private void LoadAndShowOwnedWearables(UserProfile userProfile)
...
+            if (response.success)
+                namesResult = response.response.Names.ToArray();
+            else
+                Debug.LogError("Error requesting names lambdas!");
+
+            view.SetCollectibleNames(namesResult);
+            view.SetCollectibleNamesLoadingActive(false);
+        }
+
+        private async UniTask LoadAndShowOwnedLandsAsync(UserProfile userProfile)
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs:144]: Add `CancellationToken`

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]
There is a misunderstanding of async flow and Cancellation Tokens usage. Please make fixes according to the comments

**Review by mikhail-dcl — APPROVED** [review]

**Review by kwssbocian — APPROVED** [review]
Approved for release from QA side

**Pull request merged**